### PR TITLE
Improve explanation of code coverage in excluded branches

### PR DESCRIPTION
**Pull request by edrevo**
### What does this PR do? What is the motivation?

Add some context on how code coverage works on excluded branches.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/intelligent_test_runner/_index.md [hunk 1]
```diff
@@ -65,6 +65,9 @@ Due to the [limitations](#out-of-the-box-configuration-limitations) described ab
 
 If there are other branches you want to exclude, add them on the Test Service Settings page. The query bar supports using the wildcard character `*` to exclude any branches that match, such as `release_*`.
 
+Excluded branches still collect per-test code coverage, so you might notice a performance impact on your excluded branches. However, code coverage is enabled
```
Comment (KickingTheTV) [inline content/en/intelligent_test_runner/_index.md:68]: Excluded branches collect per-test code coverage, which will have a performance impact on the total testing time. However, this performance impact is mitigated by only collecting code coverage when Datadog detects that running with code coverage will generate enough new coverage information that it offsets the cost of collecting the coverage.

**Review by KickingTheTV — CHANGES_REQUESTED** [review]
overall LGTM, but added a few modifications

**Comment by edrevo** [comment]
Wording changed.

File: content/en/intelligent_test_runner/_index.md [hunk 2]
```diff
@@ -65,6 +65,8 @@ Due to the [limitations](#out-of-the-box-configuration-limitations) described ab
 
 If there are other branches you want to exclude, add them on the Test Service Settings page. The query bar supports using the wildcard character `*` to exclude any branches that match, such as `release_*`.
 
+Excluded branches collect per-test code coverage, which will have a performance impact on the total testing time. However, this performance impact is mitigated by only collecting code coverage when Datadog detects that running with code coverage will generate enough new coverage information that it offsets the cost of collecting the coverage. You can check whether a test session has code coverage enabled or not by looking at the `@test.code_coverage.enabled` field.
```
Comment (buraizu) [inline content/en/intelligent_test_runner/_index.md:68]:
```suggestion
Excluded branches collect per-test code coverage, which has a performance impact on the total testing time. However, this performance impact is mitigated by only collecting code coverage when Datadog detects that running with code coverage generates enough new coverage information that it offsets the cost of collecting the coverage. You can check whether a test session has code coverage enabled or not by looking at the `@test.code_coverage.enabled` field.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Looks good overall, just requesting a couple of minor wording changes per the [documentation style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#tense)

**Review by buraizu — APPROVED** [review]
LGTM

**Review by KickingTheTV — APPROVED** [review]

**Comment by edrevo** [comment]
@buraizu, i'm getting a weird error in CI: https://app.datadoghq.com/logs?query=ci.pipeline_correlation_id%3A2297adb264c4e345%20ci.job_correlation_id%3Ad1df7fade70b89b1%20&cols=%40ci.job.name&event=AgAAAY1T4P0q0v6RbAAAAAAAAAAYAAAAAEFZMVQ0TGgtQUFDZWFkWGczQWdyQUFBQQAAACQAAAAAMDE4ZDUzZTktMGM0My00ODZlLWE0YjAtMzNjYTM5NzdmMmE2&index=&messageDisplay=inline&refresh_mode=paused&stream_sort=time%2Cdesc&viz=stream&from_ts=1706507984000&to_ts=1706510130000&live=false

do you know if this is something related to the build itself or my PR?

**Comment by edrevo** [comment]
@buraizu , the CI is now green. Can you merge? Thanks!

**Pull request merged**